qiskit_optimization/algorithms/warm_start_qaoa_optimizer.py

```diff
@@ -0,0 +1,339 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# todo: update documentation
+"""Implementation of the warm start QAOA optimizer."""
+
+import copy
+from abc import ABC
+from typing import Optional, List, Union, Dict, Tuple, cast
+
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.algorithms import QAOA
+from qiskit.circuit import Parameter
+
+from qiskit_optimization import QiskitOptimizationError, QuadraticProgram
+from qiskit_optimization.algorithms import MinimumEigenOptimizationResult, MinimumEigenOptimizer, \
+    OptimizationAlgorithm, OptimizationResultStatus, SolutionSample
+from qiskit_optimization.converters import QuadraticProgramConverter
+from qiskit_optimization.problems import VarType
+
+
+class BaseAggregator(ABC):
+    """A base abstract class for aggregates results"""
+
+    def aggregate(self, results: List[MinimumEigenOptimizationResult]) \
```

Comment:
Ought to have `@abtractmethod` decorator in this class I think

Reply:
Added.